### DISCUSSION TITLE
Configuration for Renovate Bot

### DISCUSSION THREAD

**Discussion by SudharakaP** [renovatebot/renovate#10609] [Request Help] [answered]
I've used the renovate bot app and it created default configurations in each of my repos which looks like this; 

```
{
  "extends": ["config:base"]
}
```
I like to move this to the org level so that I can keep a common config rather than having this file in each repo. So now if I just move this to a `renovate-config` repo at the organization level and delete this file from all repos, would that work seamlessly? 🤔

**Comment by viceice** [comment, marked as answer]
Nope, you need a config in each repo, but you can have a shared organization preset.

https://docs.renovatebot.com/config-presets/

**Reply by SudharakaP** [reply]
@viceice : Thanks much for your reply; appreciate it. So my understanding is that for example to add `preserveSemverRanges` default preset I need to add this to each config at every repo; only the presets I create (custom presets) that needs to be shared should be added to the organizational config? I am correct? 😄 🐨

**Reply by viceice** [reply]
You should add all config options which should be shared to the org preset. You can even extend the base preset there too.

In each repo only extend the shared org preset.

You can also create a org renovate config, which renovate automatically uses as onboarding preset. We do this at our org.

**Reply by rarkins** [reply]
- create `github.com/yourorg/renovate-config` with your shared config (it can extend config:base plus add other config)
- change any existing repos from extending config:base to extending `local>yourorg/renovate-config`
- any future onboarded repos should automatically get your custom org preset proposed in onboarding and not config:base

**Reply by SudharakaP** [reply]
Thanks @rarkins, Would this be the same as defining the renovate config within the `github.com/yourorg/.github` repo? And then extending `github>opencollective/.github` in the existing repos? I was just wondering as I was going to do it that way. 🤔

**Reply by rarkins** [reply]
Here's the relevant source code: https://github.com/renovatebot/renovate/blob/fad0926cb6faa2ff3a0fd4183de37f2f1a008048/lib/workers/repository/onboarding/branch/config.ts#L23-L56

So it would need to be a file `renovate-config.json` within your `.github` and would appear as `local>opencollective/.github:renovate-config` in the onboarding extends.

Note: this is so that you can achieve *automatic* onboarding PRs with your preferred preset. If you only have a few repos and can update the `extends` array manually then you can name it and locate it anywhere you like, it's just that Renovate won't be able to know to insert it into any new onboarding PRs.

**Reply by SudharakaP** [reply]
@rarkins : Thanks much for the insight. I've went along with your suggestion with a dedicated `renovate-config` repo. Now everything seems to work fine. 😄
